test/instrumentation/modules/@aws-sdk/client-sns.test.js

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict';
+
+// Test SNS instrumentation of the '@aws-sdk/client-sns' module.
+//
+// Note that this uses localstack for testing, which mimicks the SNS API but
+// isn't identical. Some known limitations:
+// TODO: check if there are limitations
+
+const semver = require('semver');
+if (process.env.GITHUB_ACTIONS === 'true' && process.platform === 'win32') {
+  console.log('# SKIP: GH Actions do not support docker services on Windows');
+  process.exit(0);
+}
+if (process.env.ELASTIC_APM_CONTEXT_MANAGER === 'patch') {
+  console.log(
+    '# SKIP @aws-sdk/* instrumentation does not work with contextManager="patch"',
+  );
+  process.exit();
+}
+if (semver.lt(process.version, '14.0.0')) {
+  console.log(
+    `# SKIP @aws-sdk min supported node is v14 (node ${process.version})`,
+  );
+  process.exit();
+}
```

Comment:
nit: Ah, `runTestFixtures` is being used. You *could* remove the guard here and add a `versionRanges: { node: '>=14' }` on each of the test cases below, if you like.

Reply:
done!